scylla-cql/src/types/serialize/row.rs

```diff
@@ -0,0 +1,47 @@
+use std::sync::Arc;
+
+use crate::frame::response::result::ColumnSpec;
+use crate::frame::value::ValueList;
+
+use super::SerializationError;
+
+pub struct RowSerializationContext<'a> {
+    columns: &'a [ColumnSpec],
+}
+
+impl<'a> RowSerializationContext<'a> {
+    pub fn columns(&self) -> &'a [ColumnSpec] {
+        self.columns
+    }
+
+    // TODO: change RowSerializationContext to make this faster
```

Comment:
Nit: what do you mean by this comment?

Reply:
Apart from list of clumns we can also keep a hashmap from name to column spec so that name lookups are in constant time instead of linear.

Reply:
I see.